include/revng/Model/CallSitePrototype.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Model/Type.h"
+#include "revng/Support/MetaAddress.h"
+#include "revng/Support/MetaAddress/YAMLTraits.h"
+#include "revng/TupleTree/TupleTree.h"
+
+/* TUPLE-TREE-YAML
+name: CallSitePrototype
+doc: Prototype of a callsite
+type: struct
+fields:
+  - name: CallerBlockAddress
+    doc: Address of the basic block of the call
+    type: MetaAddress
+  - name: Prototype
+    doc: Protoype
+    type: model::TypePath
+    optional: true
+key:
+  - CallerBlockAddress
+TUPLE-TREE-YAML */
+
+#include "revng/Model/Generated/Early/CallSitePrototype.h"
+
+class model::CallSitePrototype : public model::generated::CallSitePrototype {
+public:
+  using generated::CallSitePrototype::CallSitePrototype;
+};
```

Comment:
Add `dump` and `verify` just as all the other parts of the model.
`verify` must be called by `Binary::verify`.

Reply:
`FunctionMetadata::verify` can assume the whoel `model::Binary` verifies correctly.
`CallSitePrototype::verify` must not rely on `FunctionMetadata`, that's out of the model.
I think the logic you're describing is fine in `CallEdge::verify`.

Reply:
I'm not sure I've understood correctly. `Prototype` has been removed from `CallEdge::verify`, therefore now checking for prototype's validity ought to be done in `CallSitePrototype::verify`. Though, we need the `Start` MetaAddress of each `efa::BasicBlock` that is performing a direct/indirect call and the type of `Edge` for `CallSitePrototype::verify`, so I assume this may be done in `FunctionMetadata::verify` since in `model:Binary` we do not have the CFG anymore.